### ISSUE TITLE
Rule on sort conflicts with enum values rule

### ISSUE THREAD

**Issue by tkrop** [zalando/restful-api-guidelines#572]
The rule [sort](https://opensource.zalando.com/restful-api-guidelines/#sort) conflicts with the rule of defining [upper snake case enum values](https://opensource.zalando.com/restful-api-guidelines/#240) by suggesting to use the [lower snake case property names](https://opensource.zalando.com/restful-api-guidelines/#118) in conjunction with `+` and `-`.

**Comment by vadeg** [comment]
The may apply to `_embed` parameter [Rule 158: SHOULD allow optional embedding of sub-resources](https://opensource.zalando.com/restful-api-guidelines/index.html#158).

**Comment by ePaul** [comment]
Just to add an example:
```
parameters:
  sort:
   schema:
      type: array
      items:
        type: string
        enum:
         - "+id"
         - "-id"
         - "+name"
         - "-name"
```

**Comment by ePaul** [comment]
Current enum rule:

> MUST declare enum values using UPPER_SNAKE_CASE format [240]
> Enum values (using enum or x-extensible-enum) need to consistently use the upper-snake case format, e.g. VALUE or YET_ANOTHER_VALUE. This approach allows to clearly distinguish values from properties or other elements.

Possible new text:

> SHOULD declare enum values using UPPER_SNAKE_CASE format [240]
> Enum values (using enum or x-extensible-enum) need to consistently use the upper-snake case format, e.g. VALUE or YET_ANOTHER_VALUE. This approach allows to clearly distinguish values from properties or other elements.
> 
> This does not apply in cases where the exact values are constrained by something else, like codes defined in outside standards or using an enum to describe possible field names in sort/filter/embed parameters.


Alternatively we could propose to not actually use an enum in those cases.

**Comment by ePaul** [comment]
**From the meeting**

* We change MUST to SHOULD
* add a sentence with an example where UPPER_SNAKE doesn't work
* also move rules 240 and 125 together

@ePaul will create a proposal.

**Comment by ePaul** [comment]
Implemented by #606.
